cudax/include/cuda/experimental/__stf/internal/parallel_for_scope.cuh

```diff
@@ -60,59 +63,367 @@ __global__ void loop(const _CCCL_GRID_CONSTANT size_t n, shape_t shape, F f, tup
 
   // Help the compiler which may not detect that a device lambda is calling a device lambda
   CUDASTF_NO_DEVICE_STACK
-  auto explode_args = [&](auto... data) {
+  auto explode_args = [&](auto&&... data) {
+    CUDASTF_NO_DEVICE_STACK
+    auto const explode_coords = [&](auto&&... coords) {
+      f(coords..., data...);
+    };
     // For every linearized index in the shape
     for (; i < n; i += step)
     {
-      CUDASTF_NO_DEVICE_STACK
-      auto explode_coords = [&](auto... coords) {
-        f(coords..., data...);
-      };
       ::std::apply(explode_coords, shape.index_to_coords(i));
     }
   };
   ::std::apply(explode_args, mv(targs));
 }
 
+// Create SelectType Using Partial Specialization
+template <typename Oi, typename Ai>
```

Comment:
Nitpick:
I would prefer some more descriptive names of the types here

Reply:
@andralex was puzzled by it too, it means it should be improved or at least to make better comments if the naming makes the code too heavy